### PR TITLE
Rest api

### PR DESCRIPTION
**Pull request by frederike-ramin**

### CODE REVIEW

File: src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+     * @param entityTypeId - the id of the entity type to be searched for
+     * @param fromIndex - the index of the start of the entities to be searched for
+     * @param toIndex - the index of the end of the entities to be searched for
+     * @return - the URI to retrieve events from
+     */
+    static String getEventsOfEntityUri(long entityId, long entityTypeId, int fromIndex, int toIndex) {
+        return getEventsOfEntityBaseUri()
+                .replaceAll(getEntityIdParameter(true), Objects.toString(entityId, "0"))
+                .replaceAll(getEntityTypeIdParameter(true), Objects.toString(entityTypeId, "0"))
+                .replaceAll(getIndexFromParameter(true), Objects.toString(fromIndex, "0") )
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpoint.java]: MINOR ')' is preceded with whitespace. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AParen_Pad_BP2016)

File: src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                .replaceAll(getIndexToParameter(true), Objects.toString(toIndex, "0") );
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpoint.java]: MINOR ')' is preceded with whitespace. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AParen_Pad_BP2016)

File: src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    /**
+     * This method returns all attributes of an eventType
+     * @param request - Spark defined parameter containing request object
+     * @param response - Spark defined parameter containing response object
+     * @return - returns a JSON String of the attributes
+     */
+    String getEntityTypeAttributes(Request request, Response response);
+
+    /**
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpoint.java]: MAJOR First sentence should end with a period. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadoc_Style_BP2016)

File: src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+     * This method returns the hierarchy overview of all entityTypes.
+     * @param request - Spark defined parameter containing request object
+     * @param response - Spark defined parameter containing response object
+     * @return - returns a JSON String of the hierarchy
+     */
+    String getEntityTypeHierarchy(Request request, Response response);
+
+    /**
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpoint.java]: MAJOR First sentence should end with a period. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadoc_Style_BP2016)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+     * @return - returns a JSON string of the entity mappings of an event type
+     */
+    String getEventTypeEntityMappings(Request request, Response response);
+
+    /**
+     * This method returns the basic URI to retrieve all event types.
+     * @return - the URI to retrieve all event types from
+     */
+    static String getEventTypesBaseUri() {
+        return Argos.getRoutePrefix() + "/api/eventtypes";
```
Comment (Johannes-Schneider) [inline src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpoint.java]: Please use String.format(...)

**Review by Johannes-Schneider — CHANGES_REQUESTED** [review]

**Review by Johannes-Schneider — APPROVED** [review]

**Pull request merged**